routing/bandwidth.go

```diff
@@ -89,30 +107,71 @@ func (b *bandwidthManager) getBandwidth(cid lnwire.ShortChannelID,
 		return 0
 	}
 
-	// If our link isn't currently in a state where it can  add another
-	// outgoing htlc, treat the link as unusable.
-	if err := link.MayAddOutgoingHtlc(amount); err != nil {
-		log.Warnf("ShortChannelID=%v: cannot add outgoing htlc: %v",
-			cid, err)
+	if len(blobs) == 0 {
+		// If our link isn't currently in a state where it can add
+		// another outgoing htlc, treat the link as unusable.
+		if err := link.MayAddOutgoingHtlc(amount); err != nil {
+			log.Warnf("ShortChannelID=%v: cannot add outgoing "+
+				"htlc: %v", cid, err)
+			return 0
+		}
+
+		// Otherwise, we'll return the current best estimate for the
+		// available bandwidth for the link.
+		return link.Bandwidth()
+	}
+
+	// If the provided tlv blobs are not of the expected number, then we'll
+	// log a warning and return zero bandwidth.
+	if len(blobs) != 2 {
+		log.Warnf("ShortChannelID=%v: invalid number of tlv blobs: %v",
+			cid, len(blobs))
+		return 0
+	}
+
+	htlcTlv := blobs[0]
+	channelBlob := blobs[1]
+
+	res := false
+
+	// If the traffic shaper is present, we'll check if the payment should
+	// be carried by the channel.
+	b.trafficShaper.WhenSome(func(ts TlvTrafficShaper) {
+		res = ts.ShouldCarryPayment(amount, htlcTlv, channelBlob)
+	})
+
+	if !res {
+		log.Warnf("ShortChannelID=%v: payment not allowed by traffic "+
+			"shaper", cid)
 		return 0
 	}
 
-	// Otherwise, we'll return the current best estimate for the available
-	// bandwidth for the link.
```

Comment:
I think we might actually want to return sort of "infinite" bandwidth here instead.

Reply:
hmm given that this method is also used in `getOutgoingBalance` should I provide nil blobs when coming from that path?

Reply:
This is solved by the current version by returning the bandwidth the external traffic shaper returns.